impeller/geometry/geometry_unittests.cc

```diff
@@ -2007,14 +2020,14 @@ TEST(GeometryTest, CubicPathComponentPolylineDoesNotIncludePointOne) {
 }
 
 TEST(GeometryTest, PathCreatePolyLineDoesNotDuplicatePoints) {
-  Path path;
-  path.AddContourComponent({10, 10});
-  path.AddLinearComponent({10, 10}, {20, 20});
-  path.AddLinearComponent({20, 20}, {30, 30});
-  path.AddContourComponent({40, 40});
-  path.AddLinearComponent({40, 40}, {50, 50});
+  PathBuilder builder;
+  builder.MoveTo({10, 10});
+  builder.AddLine({10, 10}, {20, 20});
+  builder.AddLine({20, 20}, {30, 30});
```

Comment:
`Add*` inserts a new contour -- this actually needs to be `LineTo` to be equivalent.

```cpp
builder.MoveTo({10, 10});
builder.AddLine({20, 20});
builder.AddLine({30, 30});
```

And no, you're not crazy. Yes, the linear component has redundant data.

Reply:
Aghh, good catch. Thank you